### PR TITLE
Add Node-RED submenu with factory reset

### PR DESCRIPTION
**Pull request by chriadam**
Fixes issue #1462

### CODE REVIEW

**Comment by chriadam** [comment]
@ReinvdZee any chance you could help me test this one?  I did: Settings -> Firmware -> Online Updates -> Image Type = Large, then performed a firmware update.  Now the Settings page has a "Venus OS Large Features" entry, and I can see the Node Red option and interact with it.  But I'm not sure if it's doing what it's supposed to do, since there's no feedback after I trigger the reset.

File: pages/settings/PageSettingsNodeRed.qml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			ListButton {
+				id: resetButton
+				//% "Node-RED factory reset"
+				text: qsTrId("page_settings_nodered_factory_reset")
+				secondaryText: CommonWords.press_to_reset
+				allowed: true
+				onClicked: Global.dialogLayer.open(factoryResetConfirmationDialogComponent)
+
+				VeQuickItem {
+					id: nodeRedFactoryResetItem
```
Comment (DanielMcInnes) [inline pages/settings/PageSettingsNodeRed.qml:37]: can this be moved into factoryResetConfirmationDialogComponent?

**Review by DanielMcInnes — APPROVED** [review]

**Comment by ReinvdZee** [comment]
@chriadam 

There is currently no feedback from venus-platform, other than node-red being disabled (`/Services/NodeRed/Mode` set to 0), and everything under `/data/home/nodered/.node-red/node_modules` deleted.

Instead of writing `1` to `/Services/NodeRed/FactoryReset`, the GUI should write the last 4 digits of the device's unique ID (`com.victronenergy.platform/Device/UniqueId`) to this field. This is to prevent resetting it easily by hand, from `dbus-spy` for example.

See [my comment](https://github.com/victronenergy/venus/issues/1293#issuecomment-2233237447) on the original issue.

File: pages/settings/PageSettingsNodeRed.qml (hunk 1 above)
Comment (chriadam) [reply pages/settings/PageSettingsNodeRed.qml:37]: the dialog will be destroyed after being accepted, and I'd prefer not to rely on VeQuickItem doing the write synchronously for it to succeed.

**Comment by chriadam** [comment]
Thanks @ReinvdZee , updated PR

**Comment by ReinvdZee** [comment]
@chriadam,

I've just tested this and it does not work. I just found out that although we initially agreed to write the last 4 digits of the unique ID to `/Services/NodeRed/FactoryReset`, the gui needs to write just a `1` to it to initiate the reset. So it probably was correct the first time. So please revert back to writing just a `1` instead of the 4 last digits of the unique ID.

Apologies for the confusion.

**Comment by chriadam** [comment]
No problem at all, will revert

**Comment by chriadam** [comment]
Hopefully I did that correctly.  If someone could sanity check for me, that'd be greatly appreciated.

**Comment by ReinvdZee** [comment]
Tested again, works fine!

**Pull request merged**